From 801bb44aad2476d259d249b3c1d7375c7cfd8725 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 3 Nov 2011 15:51:35 +0800
Subject: [PATCH 0539/2508] ENGR00161321 [MX6 ARM2]Disable Warm reset

Current warm reset is not working with MMDC_CH1 bypass
bit set, now we disable warm reset to workaround it for
the coming release. Then, wdog reset will be cold reset.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-mx6/cpu.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx6/cpu.c b/arch/arm/mach-mx6/cpu.c
index 66dde3c..0e85d92 100644
--- a/arch/arm/mach-mx6/cpu.c
+++ b/arch/arm/mach-mx6/cpu.c
@@ -91,6 +91,13 @@ static int __init post_cpu_init(void)
 		reg |= 0x20;
 		__raw_writel(reg, base);
 	}
+
+	/* Disable SRC warm reset to work aound system reboot issue */
+	base = IO_ADDRESS(SRC_BASE_ADDR);
+	reg = __raw_readl(base);
+	reg &= ~0x1;
+	__raw_writel(reg, base);
+
 	/* Allocate IRAM for WAIT code. */
 	/* Move wait routine into iRAM */
 	cpaddr = (unsigned long)iram_alloc(SZ_4K, &iram_paddr);
-- 
1.7.7.4

